eos-payg-csv/tests/eos-payg-csv.py

```diff
@@ -0,0 +1,168 @@
+        print('eos_payg_csv:', self.__eos_payg_csv)
+        print('data dir:', self.__datadir)
+
+    def tearDown(self):
+        shutil.rmtree(self.tmpdir)
+
+    def createKey(self, contents=None):
+        if not contents:
+            contents = 'this is a key with at least 64 bytes of content ' + \
+                       'otherwise we get an error'
+        with open('key', 'w') as key_file:
+            key_file.write(contents)
+        return 'key'
+
+    def runCsv(self, *args):
+        argv = [self.__eos_payg_csv]
+        argv.extend(args)
+        print('Running:', argv)
+
+        env = os.environ.copy()
+        env['LC_ALL'] = 'C.UTF-8'
+        print('Environment:', env)
+
+        info = subprocess.run(argv, timeout=self.timeout_seconds,
+                              stdout=subprocess.PIPE,
+                              stderr=subprocess.STDOUT,
+                              env=env)
+        print('Output:', info.stdout.decode('utf-8'))
+        return info
+
+    def test_help(self):
+        """Test getting help."""
+        info = self.runCsv('-h')
+        info.check_returncode()
+        out = info.stdout.decode('utf-8').strip()
+        self.assertIn('device_id,code1,code2,code3,key\n', out)
+        self.assertIn('DEVICE_ID_1,CODE_1,CODE_2,CODE_2,KEY\n', out)
+
+    def test_help_no_args(self):
+        """Test getting help if no arguments are provided."""
+        info = self.runCsv()
+        self.assertIsNot(info.returncode, 0)
+        out = info.stdout.decode('utf-8').strip()
+        self.assertIn('usage:', out)
+        self.assertIn('following arguments are required: ACCT_KEY_CSV_FILE', out)
+
+    def test_missing_file(self):
+        """Test error handling when passing an invalid filename."""
+        info = self.runCsv('non/existing/file')
+        out = info.stdout.decode('utf-8').strip()
+        self.assertIn("failed to read ACCT_KEY_CSV_FILE", out)
+        self.assertEqual(info.returncode, 1)  # EXIT_INVALID_OPTIONS
+
+    def test_csv_wrong_columns(self):
+        """Test error handling when providing incorrectly-formatted file."""
+        info = self.runCsv(os.path.join(self.__datadir, 'wrong-columns.csv'))
+        out = info.stdout.decode('utf-8').strip()
+        self.assertIn('Provided CSV invalid:\nexpected header', out)
+        self.assertEqual(info.returncode, 1)  # EXIT_INVALID_OPTIONS
+
+    def test_csv_short_key(self):
+        """Test error handling when providing a CVS with a short key entry."""
+        info = self.runCsv(os.path.join(self.__datadir, 'short-key.csv'))
+        out = info.stdout.decode('utf-8').strip()
+        self.assertIn('Provided CSV invalid: minimum key length 64 bytes but '
+                      'CSV contains a key of 12 bytes', out)
+        self.assertEqual(info.returncode, 1)  # EXIT_INVALID_OPTIONS
+
+    def test_csv_valid(self):
+        """Test proper operation of the script for a 1-key input file."""
```

Comment:
It would be nice to test a file with more than one key, and even more than one for the same device ID, but don't consider that a blocker for merging this unless you want to

Reply:
Fixed locally.